src/main/java/com/gtnewhorizons/angelica/glsm/states/MatrixMode.java

```diff
@@ -9,6 +9,13 @@ public class MatrixMode implements ISettableState<MatrixMode> {
     protected int mode = GL11.GL_MODELVIEW;
 
     public void setMode(int mode) {
+        // Handle buggy mods
+        if(mode == GL11.GL_MODELVIEW_MATRIX) {
+            mode = GL11.GL_MODELVIEW;
```

Comment:
We need to check what the GL implementation does with bad modes. If they are ignored, such buggy code was effectively dead previously. Now, it will run and might cause undesired side effects.

Reply:
`GL_INVALID_ENUM is generated if mode is not an accepted value.`

I know kubatech (mobs info) was doing it wrong, but it was accidently doing the right thing (most likely) because MODELVIEW was the default mode.

Either we fix it, or we just silently do nothing.

Reply:
Silently doing nothing seems more correct based on your excerpt.